prov/verbs/src/ep_rdm/verbs_rdm.h

```diff
@@ -90,46 +126,65 @@ struct fi_ibv_rdm_tagged_rndv_header {
 	uint32_t mem_key;
 };
 
-struct fi_ibv_rdm_tagged_unexp_rbuff {
+struct fi_ibv_rdm_tagged_extra_buff {
 	struct fi_ibv_mem_pool_entry mpe;
 	char payload[sizeof(void *)];
 };
 
 struct fi_ibv_rdm_tagged_request {
+
+	/* Accessors and match info */
+
 	struct fi_ibv_mem_pool_entry mpe;
+	/* Request can be an element of only one queue at the moment */
+	struct dlist_entry queue_entry;
+
+	struct {
+		enum fi_ibv_rdm_tagged_request_eager_state eager;
+		enum fi_ibv_rdm_tagged_request_rndv_state rndv;
+	} state;
```

Comment:
No, these two fields are dimensions of 3-d state machine handlers array, third one is event